pkg/remote/oci/resolver.go

```diff
@@ -47,65 +65,115 @@ func (o *Resolver) List() ([]remote.ResolvedObject, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Could not parse image manifest: %w", err)
 	}
+
+	if err := o.checkImageCompliance(manifest); err != nil {
+		return nil, err
+	}
+
 	contents := make([]remote.ResolvedObject, 0, len(manifest.Layers))
 	for _, l := range manifest.Layers {
 		contents = append(contents, remote.ResolvedObject{
-			Kind:       l.Annotations["cdf.tekton.image.kind"],
-			APIVersion: l.Annotations["cdf.tekton.image.apiVersion"],
-			Name:       l.Annotations["org.opencontainers.image.title"],
+			Kind:       l.Annotations[KindAnnotation],
+			APIVersion: l.Annotations[APIVersionAnnotation],
+			Name:       l.Annotations[TitleAnnotation],
 		})
 	}
 
 	return contents, nil
 }
 
 func (o *Resolver) Get(kind, name string) (runtime.Object, error) {
-	img, err := o.retrieveImage()
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), o.timeout)
+	defer cancel()
+	img, err := o.retrieveImage(timeoutCtx)
 	if err != nil {
 		return nil, err
 	}
 
 	manifest, err := img.Manifest()
 	if err != nil {
-		return nil, fmt.Errorf("Could not parse image manifest: %w", err)
+		return nil, fmt.Errorf("could not parse image manifest: %w", err)
+	}
+
+	if err := o.checkImageCompliance(manifest); err != nil {
+		return nil, err
 	}
 
 	layers, err := img.Layers()
 	if err != nil {
-		return nil, fmt.Errorf("Could not read image layers: %w", err)
+		return nil, fmt.Errorf("could not read image layers: %w", err)
 	}
 
 	for idx, l := range manifest.Layers {
-		lKind := l.Annotations["cdf.tekton.image.kind"]
-		lName := l.Annotations["org.opencontainers.image.title"]
+		lKind := l.Annotations[KindAnnotation]
+		lName := l.Annotations[TitleAnnotation]
 
 		if kind == lKind && name == lName {
-			return readLayer(layers[idx])
+			obj, err := readTarLayer(layers[idx])
+			if err != nil {
+				// This could still be a raw layer so try to read it as that instead.
+				return readRawLayer(layers[idx])
+			}
+			return obj, nil
 		}
 	}
-	return nil, fmt.Errorf("Could not find object in image with kind: %s and name: %s", kind, name)
+	return nil, fmt.Errorf("could not find object in image with kind: %s and name: %s", kind, name)
 }
 
 // retrieveImage will fetch the image's contents and manifest.
-func (o *Resolver) retrieveImage() (v1.Image, error) {
+func (o *Resolver) retrieveImage(ctx context.Context) (v1.Image, error) {
 	imgRef, err := imgname.ParseReference(o.imageReference)
 	if err != nil {
 		return nil, fmt.Errorf("%s is an unparseable image reference: %w", o.imageReference, err)
 	}
-	return ociremote.Image(imgRef, ociremote.WithAuthFromKeychain(o.keychain))
+	return ociremote.Image(imgRef, ociremote.WithAuthFromKeychain(o.keychain), ociremote.WithContext(ctx))
+}
+
+// checkImageCompliance will perform common checks to ensure the Tekton Bundle is compliant to our spec.
```

Comment:
Nice! Later we might want to move this off the Resolver struct (it doesn't appear to use it at all today) and make this public/shared so it can be reused elsewhere.